### PR TITLE
[3.x] PHP CS Fixer PHPCSFixer Ruleset

### PR DESCRIPTION
**Pull request by ravage84**
Type: refactoring 
Issue: Resolves None
Breaking change: No

Depends on:
- https://github.com/phpmd/phpmd/pull/1139

This adds & applies the full "PHP CS  Fixer" ruleset but amends some rules.

Many rules are redundant and kept only for discussion & understandability, including their description line.

The config file contains a few TODOs. Some rules were applied already (see separte commits), for some we would need to make a decision.

Also, I added a Composer script for executing a PHP CS Fixer check just for conventience.
We don't need to keep them long term or can rename it.

@AJenbo this is not yet merged with your list. 
Will do that in a PR that is based on this one.

### CODE REVIEW

**Comment by AJenbo** [comment]
oh nice, nullable_type_declaration_for_default_null_value just fixed our PHP 8.4 issues :D

**Comment by ravage84** [comment]
> oh nice, nullable_type_declaration_for_default_null_value just fixed our PHP 8.4 issues :D

Yeah, thought so, too 😏

**Review by AJenbo — APPROVED** [review]
love it, we just need to merge https://github.com/phpmd/phpmd/pull/1135 so that it doesn't block everything else.

**Comment by AJenbo** [comment]
> > oh nice, nullable_type_declaration_for_default_null_value just fixed our PHP 8.4 issues :D
> 
> Yeah, thought so, too 😏

I have actually see the PHP team sometimes use php-cs-fixer as justification when they deprecate things. I wonder if it was also the case here.

Arh indeed they did: https://wiki.php.net/rfc/deprecate-implicitly-nullable-types

File: src/main/php/PHPMD/TextUI/CommandLineOptions.php [hunk 1]
```diff
@@ -251,107 +251,158 @@ public function __construct(array $args, array $availableRuleSets = [])
                 case '--':
                     $this->refuseValue($equalChunk);
                     $listenOptions = false;
+
                     break;
+
                 case '--verbose':
```
Comment (AJenbo) [inline src/main/php/PHPMD/TextUI/CommandLineOptions.php:257]: I probably would have only gone for a blank line after and not also before the break.

File: .php-cs-fixer.php [hunk 2]
```diff
@@ -93,9 +94,268 @@
...
+            ]
+        ],
+        // The namespace declaration line shouldn't contain leading whitespace.
+        'no_leading_namespace_whitespace' => true,
+        // Either language construct `print` or `echo` should be used.
+        'no_mixed_echo_print' => true,
+        // Operator `=>` should not be surrounded by multi-line whitespaces.
+        'no_multiline_whitespace_around_double_arrow' => true,
+        // Properties MUST not be explicitly initialized with `null` except when they have a type declaration (PHP 7.4).
+        'no_null_property_initialization' => true, // TODO Arguable, could remove info about nullability
```
Comment (AJenbo) [inline .php-cs-fixer.php:218]: Lets disable this for now, we can revisit it once we have native types.
```suggestion
        'no_null_property_initialization' => false, // TODO Arguable, could remove info about nullability
```

**Comment by AJenbo** [comment]
Looks like this was replaced by https://github.com/phpmd/phpmd/pull/1138

**Pull request closed without merging**